pkg/util/kubernetes/kubelet/types_kubelet.go

```diff
@@ -146,3 +146,14 @@ type ContainerStateTerminated struct {
 	StartedAt  time.Time `json:"startedAt"`
 	FinishedAt time.Time `json:"finishedAt"`
 }
+
+// KubeletSpec contains fields for unmarshalling a Kubelet Spec
+type KubeletSpec struct {
+	NumCores        int `json:"num_cores"`
+	CpuFrequencyKHz int `json:"cpu_frequency_khz"`
+	MemoryCapacity  int `json:"memory_capacity"`
+	// Some other fields not captureds (yet)
+	CloudProvider string `json:"cloud_provider"`
+	InstanceType  string `json:"instance_type"`
```

Comment:
Should I try to decode as many fields as possible (and fill the `Some other fields not captured` stuff) ?
Or should I rather try to decode only the fields I'm interested in (and remove everything but the `InstanceId` one) ?

Reply:
I don't think it is necessary for your fix. But maybe it can interest @mfpierre for the `container orchestration view`

Reply:
@L3n41c only decode the fields you're interested in to lower the memory footprint of the agent.
For full pod collection I'll be adding a new method returning raw kubernetes objects https://github.com/DataDog/datadog-agent/blob/9250aa848793062820e26d6bae4d090fe760f8ab/pkg/util/kubernetes/kubelet/kubelet.go#L177-L198